app/http.php

```diff
@@ -86,50 +90,57 @@
 App::setResource('balancer', function (Algorithm $algorithm, Request $request) {
     global $state;
     $runtimeId = $request->getHeader('x-opr-runtime-id', '');
+    $method = $request->getHeader('x-opr-addressing-method', ADDRESSING_METHOD_ANYCAST_EFFICIENT);
 
     $group = new Group();
 
+    if ($method === ADDRESSING_METHOD_ANYCAST_FAST) {
+        $algorithm = new Random();
+    }
+
     // Cold-started-only options
     $balancer1 = new Balancer($algorithm);
 
     // Only online executors
     $balancer1->addFilter(fn ($option) => $option->getState('status', 'offline') === 'online');
 
-    // Only low host-cpu usage
-    $balancer1->addFilter(function ($option) {
-        /**
-         * @var array<string,mixed> $state
-         */
-        $state = \json_decode($option->getState('state', '{}'), true);
-        return ($state['usage'] ?? 100) < 80;
-    });
-
-    // Only low runtime-cpu usage
-    if (!empty($runtimeId)) {
-        $balancer1->addFilter(function ($option) use ($runtimeId) {
+    if ($method === ADDRESSING_METHOD_ANYCAST_EFFICIENT) {
+        // Only low host-cpu usage
+        $balancer1->addFilter(function ($option) {
             /**
              * @var array<string,mixed> $state
              */
             $state = \json_decode($option->getState('state', '{}'), true);
+            return ($state['usage'] ?? 100) < 80;
```

Comment:
is the usage is empty why do we set it to 100 ? and not 0 ?

Reply:
uknown usage probably means health endpoint didnt report. Chances are he didn't respond because of bad state, not good state.
So we want to consider that "worst" option.

Reply:
okay sounds good